### PR TITLE
Format all code using cargo fmt

### LINKED ISSUE #235: Rustfmt
`cargo fmt --all -- --check` should run in a Travis matrix build with a pinned nightly version and builds should fail when . The nightly should be pinned because:

1. the version of Rust determines the version of rustfmt when installing via the standard `rustup component add rustfmt-preview`
2. if you don't pin a nightly then:
  1. in order to pass CI you might have to let rustfmt modify code you didn't touch, making for an unclear diff.
  2. it will randomly turn out the `master` branch is "broken" if you re-run the CI when a new nightly/rustfmt version comes out.

The pinned nightly should be periodically updated. We should document in a `CONTRIBUTING.md` as well as in a [PR template](https://help.github.com/articles/creating-a-pull-request-template-for-your-repository/) that the author should have run rustfmt and that they should not include a separate rustfmt commit which will clog up our git history. The PR template can include a lil' checkbox they should tick to indicate they're code has been formatted with the correct nightly.

### PR DESCRIPTION
**Pull request by dnaq**
This is a change that has been long overdue and it should simplify for
others wanting to contribute to the repo.

Unfortunately it touches a lot of files, which in turn might lead to
some unnecessary merge conflicts for existing pull-requests. On the
other hand it's not really that hard for those contributors to run cargo
fmt on their code as well.

Closes #235

### CODE REVIEW

**Comment by Dylan-DPC** [comment]
Yes but then we will have conflicts depending on the version of nightly. Is this run on latest nightly?

**Comment by dnaq** [comment]
rustc 1.32.0-nightly (6b9b97bd9 2018-11-15)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -67,7 +67,8 @@ their personal fork and create pull requests to bring those changes into the
 source repository.
 
 Ensure that your changes conform to how the rest of the crate is written and follows a 
-similar API, patterns, namings & other conventions. 
+similar API, patterns, namings & other conventions. Also ensure that your code is formatted
+using [rustfmt].
```
Comment (kpp) [inline CONTRIBUTING.md]: No. There must be a special check in CI.

Comment (kpp) [inline CONTRIBUTING.md]: No. There must be a special check in CI.

Comment (Dylan-DPC) [reply CONTRIBUTING.md]: Yah but CI doesn't fix the code,it will only check it, the user still has to run `fmt`

Comment (kpp) [reply CONTRIBUTING.md]: OK. I agree.

File: appveyor.yml [hunk 2]
```diff
@@ -23,6 +23,7 @@ install:
       appveyor-retry appveyor DownloadFile https://win.rustup.rs/ -FileName rustup-init.exe
       cmd.exe /c .\rustup-init.exe -y --default-host "$env:TARGET" --default-toolchain "$env:RUST_CHANNEL" 2`>`&1
       $env:PATH += ";$env:USERPROFILE\.cargo\bin"
+      rustup component add rustfmt-nightly
```
Comment (kpp) [inline appveyor.yml]: No need to add the check to Win build

File: .travis.yml [hunk 3]
```diff
@@ -2,12 +2,15 @@ os:
...
 language: rust
+before_script:
+  - rustup component add rustfmt-preview
 rust:
   - stable
   - beta
   - nightly
 sudo: false
 script:
+  - cargo fmt --all -- --check
```
Comment (Dylan-DPC) [inline .travis.yml]: this is wrong since it runs fmt on both stable & nightly. Fmt should be run only on nightly

Comment (dnaq) [reply .travis.yml]: How do you do that in travis?

Comment (kpp) [reply .travis.yml]: `if [ "$TRAVIS_RUST_VERSION" = "nightly" ]; then BLAHBLAH; fi`

**Comment by kpp** [comment]
Also it would be good to rebase on master and run rustfmt again before telling bors to merge it.

File: .travis.yml (hunk 3 above)
Comment (Dylan-DPC) [reply .travis.yml]: no you don't need to do that. You have to create separate matrices. Look at [this](https://github.com/uuid-rs/uuid/blob/master/.travis.yml) for an example.

Comment (dnaq) [reply .travis.yml]: @Dylan-DPC do you have the time to do these changes?, I won't have time do do anything now for a while.

**Comment by Dylan-DPC** [comment]
bors: r+

**Pull request merged**